### ISSUE TITLE
Remving Title Links from Joomla Article Particle

### ISSUE THREAD

**Issue by dmleeman** [gantry/gantry5#2820]
Joomla 3.9.27
PHP 7.4.16
Gantry 5.4.37

I'm having some problem with Joomla and Gantry5 Joomla Articles particle. I have set up Modules in which the Joomla Article particle sits. Each Article has its own title and this is shown as the title of the article in the module/particle combo.
The problem is that I cannot remove the link from the article title, even through use of Joomla globals. So, all articles try to open on the home page as itemID101 from the link inherent in the title.
Has anyone come across this?
I am only using the default Joomla Articles particle that comes with base Gantry5

![image](https://user-images.githubusercontent.com/40118574/120842244-5a75c080-c564-11eb-9a90-0525abcc4248.png)

**Comment by mahagr** [comment]
Particles generally use their own settings as they are independent of Joomla components and modules. After a quick look, I don't think there's an option to do this in the particle; maybe we need to add one?

**Comment by dmleeman** [comment]
Given that Joomla itself allows the choice of linked titles then I would be a big fan of any change to allow the choice in such a fundamental Particle. The current method can seriously mess with SEO as Google bots try to follow the links. It's an avoidable situation.
Even an override would do for now.

**Comment by mahagr** [comment]
Be free to make a pull request about this. :)

**Comment by dmleeman** [comment]
Ah, yes :) I'm in the wrong place again. Thank you Matias

**Comment by mahagr** [comment]
Seriously, if you find a way to improve Gantry, all contributions are welcome.

In this case, the easiest way is just to add an option into the blueprint to support this feature.

**Comment by dmleeman** [comment]
Thanks Matias, will do. Still getting used to github and it's structure for such requests. So will deal with my In Tray and look at this later today.

**Comment by dmleeman** [comment]
I think it is contentarray.html.twig in /media/gantry5/engines/nucleus/particles

{% if display.title.enabled %}
                                    <div class="g-array-item-title">
                                        <h3 class="g-item-title">
                                            <a href="{{ article.route }}">
                                                {{ display.title.limit ? article.title|truncate_text(display.title.limit) : article.title }}
                                            </a>
                                        </h3>
                                    </div>
                                {% endif %}
So, it is a case of removing the a href link on line 72 and 74

What do you think is the safest syntactically correct method? Simple comment?

**Comment by mahagr** [comment]
I would just add an option for title link and display either `<a>` or `<span>` depending on the new setting.

**Comment by dmleeman** [comment]
Ok, am new to TWIG so will try to work out the syntax this week. Helps me learn...

**Comment by marktaylor46** [comment]
Just a note here - the link should not just be removed as many people want it. Instead an additional setting should be made in the YAML to enable/disable the link and the link should be set/not set on that basis - also ensure that the default is set to "enabled" to ensure backwards compatibility with existing sites.

Regards, Mark.

**Comment by dmleeman** [comment]
Yes, I was originally just going for a solution that would work in 'my own bubble', but you're right. I am still new to yaml and TWIG but this look s a good manageable place to start on providing a working solution that others may find useful too.

**Comment by marktaylor46** [comment]
Just make sure that you do testing on an existing site that doesn't have the new field with your new yaml and twig to make sure that it behaves the same way as now when the field doesn't exist in existing sites (the new field won't exist and won't have a value until the "contentarray" particle has been resaved).

**Comment by dmleeman** [comment]
I forgot about re-saving the Particle - that makes sense (I guess the original options are saved to database?). I can test it on my own site too.

Thanks Mark

**Comment by marktaylor46** [comment]
Nothing is saved in the database for a particle. - it's store in flat files. What you have to do is test for the field having a value and if it doesn't treat it as if it's set to enabled. You cannot assume that everyone will resave the particle because they won't.

**Comment by dmleeman** [comment]
Particles store in flat files - ok, well I've learned that today if nothing else. You're right about forgetting to re-save Particles because that's exactly what I did too.

**Comment by marktaylor46** [comment]
Just use the |default filter on the field to default it to "enabled" if no value is set when you are testing the field. Look at other particle there are plenty of examples of this.

**Comment by mahagr** [comment]
FYI: Particles in modules and widgets are an exception for the rule -- they are stored into the DB. :)

**Comment by dmleeman** [comment]
Thanks Matias, yes I present almost all of my working Particles inside Joomla modules in order to reduce the number of layouts needed and then use the native Joomla menu assignment to control their display on pages so that is good. It makes sense that those would be wholly stored in db, otherwise it would be a structure of a db stored module calling in flat file Particle and that would be clearly less efficient.
It is important to me that you made this clear distinction. Much appreciated.

**Comment by marktaylor46** [comment]
Yes, I deliberately didn't mention that so as not to confused the point but it amounts to the same thing the yaml information is store in json in the database for widget and modules where the module/widget has been put inside a Gantry 5 particle wrapper.

**Comment by mahagr** [comment]
I made this an enchantment as it would be useful to have this feature, for example, if you just want to use articles/post/pages like inline modules.
